python/ray/autoscaler/_private/commands.py

```diff
@@ -737,6 +713,45 @@ def get_or_create_head_node(config: Dict[str, Any],
         cli_logger.print("  {}", remote_shell_str.strip())
 
 
+def _prepare_config_for_head_node(config, provider, no_restart):
```

Comment:
can you add type annotation and function level description?
Any idea for a better name since we already have provider.prepare_for_head_node?